lib/ts/test-utils.ts

```diff
@@ -0,0 +1,209 @@
+import { html, fixture, expect, unsafeStatic } from "@open-wc/testing";
+import { screen } from "@testing-library/dom";
+import { visualDiff } from "@web/test-runner-visual-regression";
+
+// TODO reinstate "theme-dark" test once we add ability to skip tests or resolve dark mode contrast issues
+// const colorThemes = ["theme-dark", "theme-light"];
+const colorThemes = ["theme-light"];
+const baseThemes = ["", "theme-highcontrast"];
+
+interface TestOptions {
+    testColorThemes: boolean;
+    testHighContrast: boolean;
+    includeNullVariant: boolean;
+    includeNullModifier: boolean;
+}
+
+interface TestModifiers {
+    primary?: string[];
+    secondary?: string[];
+    global?: string[];
+    standalone?: string[];
+}
+
+interface TestProps {
+    classes: string;
+    testid: string;
+    theme: string[];
+}
+
+const attrObjToString = (attrs: Record<string, string>): string => {
+    const attrString = Object.keys(attrs).map((key) => {
+        return `${key}="${attrs[key]}"` || "";
+    });
+    return attrString.join(" ") || "";
+};
+
+export const buildTestid = (arr: string[]) => arr.filter(Boolean).join("-");
+
+export const getTestVariations = ({
+    baseClass,
+    variants = [],
+    modifiers,
+    options = {
+        testColorThemes: true,
+        testHighContrast: true,
+        includeNullVariant: false,
+        includeNullModifier: true,
+    },
+}: {
+    baseClass: string;
+    variants?: string[];
+    modifiers?: TestModifiers;
+    options?: TestOptions;
+}) => {
```

Comment:
> Potentially add a return type for the function

👍 good call! I'll add that

> testId: string // shall we call this just id?

I think I'd like to stick with `testid`. The Testing Library function [getByTestId](https://testing-library.com/docs/queries/bytestid/#api) targets the`data-testid` attribute, so I went with `testid` (note lack of camelCasing). Also, the util function `makeTestElement` accepts attributes, so it could wind up with a collision if we ever included an `id`. 

> theme: string // maybe we can have a more restrictive type for this (union)

Another good idea! On it.